types/mongoose/index.d.ts

```diff
@@ -2453,22 +2397,22 @@ declare module "mongoose" {
      * Performs aggregations on the models collection.
      * If a callback is passed, the aggregate is executed and a Promise is returned.
      * If a callback is not passed, the aggregate itself is returned.
-     * @param ... aggregation pipeline operator(s) or operator array
+     * @param aggregations pipeline operator(s) or operator array
      */
-    aggregate(...aggregations: Object[]): Aggregate<Object[]>;
-    aggregate(...aggregationsWithCallback: Object[]): Promise<Object[]>;
+    aggregate(aggregations: any[]): Aggregate<any[]>;
```

Comment:
Can be simplified to:
```ts
aggregate(aggregations: any[], cb?: (err: any, res: any[]) => void): Promise<any[]>
```
Based on: http://mongoosejs.com/docs/api.html#aggregate_aggregate

Reply:
@simonxca Even based on the link you posted, it **cannot** be simplified.

"*If a callback is passed, the aggregate is executed and a Promise is returned. If a callback is not passed, the aggregate itself is returned.*"

Reply:
Oops! Misread the return type lol